vhdl_syntax/src/tokens/tokenizer.rs

```diff
@@ -0,0 +1,1330 @@
+        let quote = self.skip().expect("Input empty while tokenizing quoted");
+        buf.push(quote);
+        let mut found_end = false;
+        while let Some(chr) = self.skip() {
+            buf.push(chr);
+            if chr == quote {
+                if self.current == Some(quote) {
+                    buf.push(chr);
+                    self.skip();
+                } else {
+                    found_end = true;
+                    break;
+                }
+            }
+        }
+        if !found_end {
+            Unterminated
+        } else if quote == b'"' {
+            StringLiteral
+        } else if quote == b'\\' {
+            Identifier
+        } else {
+            Unknown
+        }
+    }
+
+    /// Consume a trivia piece (i.e., whitespace, newline, comments, ...)
+    fn consume_trivia_piece(&mut self) -> Option<TriviaPiece> {
+        macro_rules! count_chars {
+            ($ch:literal) => {{
+                let mut count = 0;
+                while self.skip_if_eq($ch) {
+                    count += 1;
+                }
+                count
+            }};
+        }
+        Some(match self.current? {
+            b'\t' => TriviaPiece::HorizontalTabs(count_chars!(b'\t')),
+            /*vertical tab*/
+            0x0B => TriviaPiece::VerticalTabs(count_chars!(0x0B)),
+            b'\r' => {
+                if self.peek() == Some(b'\n') {
+                    let mut count = 0;
+                    while self.current == Some(b'\r') && self.peek() == Some(b'\n') {
+                        self.skip();
+                        self.skip();
+                        count += 1;
+                    }
+                    TriviaPiece::CarriageReturnLineFeeds(count)
+                } else {
+                    TriviaPiece::CarriageReturns(count_chars!(b'\r'))
+                }
+            }
+            0x0C => TriviaPiece::FormFeeds(count_chars!(0x0C)),
+            b'\n' => TriviaPiece::LineFeeds(count_chars!(b'\n')),
+            b' ' => TriviaPiece::Spaces(count_chars!(b' ')),
+            b'-' => {
+                if self.peek() == Some(b'-') {
+                    self.skip();
+                    self.skip();
+                    let mut comment = Latin1String::default();
+                    while let Some(ch) = self.skip_if(|ch| !matches!(ch, b'\r' | b'\n')) {
+                        comment.push(ch);
+                    }
+                    TriviaPiece::LineComment(comment.to_string())
+                } else {
+                    return None;
+                }
+            }
+            b'/' => {
+                if self.peek() == Some(b'*') {
+                    self.skip();
+                    self.skip();
+                    let mut comment = Latin1String::default();
+                    loop {
+                        if self.current == Some(b'*') && self.peek() == Some(b'/') {
+                            self.next();
+                            self.next();
```

Comment:
While playing around I stumbled over this. This should probably `self.skip()` instead of `self.next()`.